### PR TITLE
W-16675934 external-processing-gr

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by juanmolle — DISMISSED** [review]
LGTM

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 1]
```diff
@@ -41,12 +41,16 @@ In Local Mode, you apply the External Processing policy to your API instance via
...
     messageTimeout: <integer> // OPTIONAL
-    requestHeaderMode: <string> // REQUIRED, default: "send"
-    responseHeaderMode: <string> // REQUIRED, default: "send"
-    requestBodyMode: <string> // REQUIRED, default: "none"
-    responseBodyMode: <string> // REQUIRED, default: "none"
+    requestHeaderMode: <string> // OPTIONAL, default: "send"
+    responseHeaderMode: <string> // OPTIONAL, default: "send"
+    requestBodyMode: <string> // OPTIONAL, default: "none"
+    responseBodyMode: <string> // OPTIONAL, default: "none"
     requestTrailerMode: <string> // OPTIONAL, default: "send"
```
Comment (juanmolle) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]: default skip

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 2]
```diff
@@ -41,12 +41,16 @@ In Local Mode, you apply the External Processing policy to your API instance via
...
     responseTrailerMode: <string> // OPTIONAL, default: "send"
```
Comment (juanmolle) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]: default skip

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 3]
```diff
@@ -41,12 +41,16 @@ In Local Mode, you apply the External Processing policy to your API instance via
   config: 
     uri: <string> // REQUIRED
     messageTimeout: <integer> // OPTIONAL
```
Comment (juanmolle) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]: you can add default : 1000

**Review by juanmolle — DISMISSED** [review]
LGTM

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 4]
```diff
@@ -60,43 +64,58 @@ In Local Mode, you apply the External Processing policy to your API instance via
...
-| Required
-| `send`
+| Optional
+| `skip`
 | Whether the HTTP response trailer headers are sent to the processing service. Supported values are `send` and `skip`.
 
+| `failureModeAllow`
+| Optional
+| `false`
+| When true, if the gRPC stream prematurely closes or cannot be open, request processing continues without error.
```
Comment (Crispy-Salesforce) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]:
```suggestion
| When `true`, if the gRPC stream prematurely closes or can’t be opened, request processing continues without error.
```

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 5]
```diff
@@ -60,43 +64,58 @@ In Local Mode, you apply the External Processing policy to your API instance via
...
+
+| `maxMessageTimeout`
+| Optional
+| 0
+| Specifies the upper bound the processing server can override `messageTimeout`.
+
+| `allowModeOverride`
+| Optional
+| `true`
+| Enables the processing server to override the policy's configuration. When true, the server can override the configuration of the `requestHeaderMode`, `responseHeaderMode`, `requestBodyMode`, `responseBodyMode`, `requestTrailerMode`, and `responseTrailerMode` parameters.
```
Comment (Crispy-Salesforce) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]:
```suggestion
| Enables the processing server to override the policy configuration. When `true`, the server can override the configuration of the `requestHeaderMode`, `responseHeaderMode`, `requestBodyMode`, `responseBodyMode`, `requestTrailerMode`, and `responseTrailerMode` parameters.
```

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 6]
```diff
@@ -60,43 +64,58 @@ In Local Mode, you apply the External Processing policy to your API instance via
...
+
 |===
 
-NOTE: Leaving a required value blank configures the parameter with the default value. No value is set for optional values left blank.
+NOTE: Leaving a value blank configures the parameter with the default value. If a required parameter does not have a default value, you must configure the parameter.
```
Comment (Crispy-Salesforce) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]:
```suggestion
NOTE: Leaving a value blank configures the parameter with the default value. If a required parameter doesn’t have a default value, you must configure the parameter. 
```

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 7]
```diff
@@ -129,20 +148,25 @@ When you apply the External Processing policy to your API from the UI, configure
 |Response body processing mode | Mode that the response body is sent in.
 |Request trailer processing mode | Whether the requests trailers are sent to the processing service.
 |Response trailer processing mode | Whether the response trailers are sent to the processing service.
+|Failure mode allow | When true, if the gRPC stream prematurely closes or cannot be open, request processing continues without error.
```
Comment (Crispy-Salesforce) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]:
```suggestion
|Failure mode allow | When `true`, if the gRPC stream prematurely closes or can’t be opened, request processing continues without error.
```

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 8]
```diff
@@ -129,20 +148,25 @@ When you apply the External Processing policy to your API from the UI, configure
...
+|Max message timeout | Specifies the upper bound the processing server can override *Message timeout*.
+|Allow mode override | Enables the processing server to override the policy's configuration. When true, the server can override the configuration of the *Request header processing mode*, *Response header processing mode*, *Request body processing mode*, *Response body processing mode*, *Request trailer processing mode*, and *Response trailer processing mode* parameters.
```
Comment (Crispy-Salesforce) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]:
```suggestion
|Allow mode override | Enables the processing server to override the policy configuration. When `true`, the server can override the configuration of the *Request header processing mode*, *Response header processing mode*, *Request body processing mode*, *Response body processing mode*, *Request trailer processing mode*, and *Response trailer processing mode* parameters.
```

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 9]
```diff
@@ -129,20 +148,25 @@ When you apply the External Processing policy to your API from the UI, configure
...
 |===
 
+NOTE: Leaving a value blank configures the parameter with the default value. If a required parameter does not have a default value, you must configure the parameter.
```
Comment (Crispy-Salesforce) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]:
```suggestion
NOTE: Leaving a value blank configures the parameter with the default value. If a required parameter doesn't have a default value, you must configure the parameter. 
```

File: gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc [hunk 10]
```diff
@@ -129,20 +148,25 @@ When you apply the External Processing policy to your API from the UI, configure
...
+. Flex Gateway forwards the initial parts of the client HTTP request that you configure to the external processing service. At this point, the external processing service can request additional parts of the request if *Allow mode override* is enabled.
 . After Flex Gateway receives the modified HTTP request, Flex Gateway sends the HTTP request to the upstream API service.
 . Flex Gateway receives the HTTP response from the upstream API service and forwards the initial parts of the HTTP response that you configure to the external processing service. At this point, the external processing service can request additional parts of the HTTP response.
 . Flex Gateway returns the modified HTTP response from the external processing service to the client.
 
-By default, if Flex Gateway cannot establish a connection to the gRPC service or if the gRPC service returns an error, the policy fails. 
+By default, if Flex Gateway cannot establish a connection to the gRPC service or if the gRPC service returns an error, the policy fails. To forward requests regardless of connection or error status, enable the *Failure mode allow* parameter.
```
Comment (Crispy-Salesforce) [inline gateway/1.8/modules/ROOT/pages/policies-included-external-processing.adoc]:
```suggestion
By default, if Flex Gateway can't establish a connection to the gRPC service or if the gRPC service returns an error, the policy fails. To forward requests regardless of connection or error status, enable the *Failure mode allow* parameter.
```

**Review by Crispy-Salesforce — DISMISSED** [review]

**Review by Crispy-Salesforce — APPROVED** [review]

**Review by marinasasso — APPROVED** [review]

**Review by marinasasso — APPROVED** [review]

**Pull request merged**